logic/minecraft/WorldList.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2015 <copyright holder> <email>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include <QList>
+#include <QString>
+#include <QDir>
+#include <QAbstractListModel>
+#include "minecraft/World.h"
+
+class QFileSystemWatcher;
+
+class WorldList : public QAbstractListModel
+{
+    Q_OBJECT
+public:
+    enum Columns {
+        NameColumn
+    };
+
+    WorldList ( const QString &dir );
+
+    virtual QVariant data ( const QModelIndex &index, int role = Qt::DisplayRole ) const;
```

Comment:
This entire file could use some clang-formatting (spaces everywhere, newlines before {, ; after function body and more)